slackButton.go

```diff
@@ -0,0 +1,58 @@
+package slackauth
+
+import (
+	"html/template"
+	"net/http"
+	"strings"
+
+	log15 "gopkg.in/inconshreveable/log15.v2"
+)
+
+const (
+	BOT      = "bot"
+	WEBHOOK  = "incoming-webhook"
+	COMMANDS = "commands"
+)
+
+// Options has all the configurable parameters for Slack button.
+type SlackButtonOptions struct {
+	// Addr is the address where the Slack button will be exposed, e.g: :8080, 0.0.0.0:8989, etc.
+	Addr string
+	//Scopes is the list of the allowed scopes
+	Scopes []string
+	//ClientId for the app bot
+	ClientId string
+	//ButtonTpl is the path to the Slack button template
+	ButtonTpl string
+}
+
+type slackButtonServer struct {
+	options  *SlackButtonOptions
+	template *template.Template
+}
+
+func CreateButtonServer(options *SlackButtonOptions) (*slackButtonServer, error) {
+	successTpl, err := readTemplate(options.ButtonTpl)
+	if err != nil {
+		return nil, err
+	}
+	buttonServer := &slackButtonServer{options, successTpl}
+
+	return nil, buttonServer
+}
+
+func (s *slackButtonServer) ServeButton() error {
+	log15.Info("Starting slack button server", "addr", s.options.Addr)
+	http.HandleFunc("/", s.buttonHandler)
```

Comment:
Instead of having 2 separate servers i'd rather have just one with the slack button at `/` and the other at `/auth`